src/components/AssetsList/index.jsx

```diff
@@ -0,0 +1,221 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+import WrappedMessage from '../../utils/i18n/formattedMessageWrapper';
+import messages from './displayMessages';
+import styles from './AssetsList.scss';
+
+const initialState = {
+  selectedAssetIndex: -1,
+  selectedAssetPage: -1,
+};
+
+export default class AssetsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      ...initialState,
+    };
+
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.onAssetClick = this.onAssetClick.bind(this);
+    this.onListBoxFocus = this.onListBoxFocus.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // if page has changed, reset the index of the selected index
+    if (this.props.assetsListMetadata.page !== nextProps.assetsListMetadata.page) {
+      if (nextProps.assetsListMetadata.page === this.state.selectedAssetPage) {
+        this.setState({
+          selectedAssetIndex: this.getSelectedAssetIndex(nextProps.assetsList,
+            nextProps.selectedAsset),
+        });
+      } else {
+        this.setState({
+          selectedAssetIndex: -1,
+        });
+      }
+    }
+
+    /*
+      if the selectedAsset is set to empty object, reset selectedAssetIndex; this is to
+      handle the case when the EditImageModal closes
+    */
+    if (Object.keys(this.props.selectedAsset).length !== 0 &&
+        Object.keys(nextProps.selectedAsset).length === 0) {
+      this.setState({
+        ...initialState,
+      });
+    }
+  }
+
+  onListBoxFocus = () => {
+    // if no list item is selected, select first item on  listbox focus
+    if (this.state.selectedAssetIndex === -1) {
+      this.setState(
+        { selectedAssetIndex: this.state.selectedAssetIndex + 1,
```

Comment:
Why not just pass `0` here? That way this `setState` call won't have to rely on existing state.